### PR TITLE
Memkind change 4.5

### PR DESCRIPTION
**Pull request by KFilipek**
This PR introduce Memkind with Jemalloc 4.5.
Warnings fixed by removing unnecessary includes.

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/redis/71)

### CODE REVIEW

**Comment by KFilipek** [comment]
unit/type/set cause problems after change, so there is a bug related with allocations.

**Review by TomekIdczak97 — APPROVED** [review]
LGTM

**Comment by michalbiesek** [comment]
Consider update the commit message "Changed Memkind to commit 4604e94a" with information about update version of memkind from 1.4.0 to 1.6.0

File: src/server.c [hunk 1]
```diff
@@ -566,7 +566,16 @@ dictType objectKeyPointerValueDictType = {
 };
 
 /* Set dictionary type. Keys are SDS strings, values are ot used. */
-dictType setDictType = {
+dictType setDictTypeZ = {
+    dictSdsHash,               /* hash function */
+    NULL,                      /* key dup */
+    NULL,                      /* val dup */
+    dictSdsKeyCompare,         /* key compare */
+    dictSdsDestructor,        /* key destructor */
```
Comment (michalbiesek) [inline src/server.c]: fix align of comment

File: src/server.c [hunk 2]
```diff
@@ -566,7 +566,16 @@ dictType objectKeyPointerValueDictType = {
...
+    NULL                       /* val destructor */
+};
+
+dictType setDictTypeM = {
```
Comment (michalbiesek) [inline src/server.c]: add description of this structure, like for example it is done for  _commandTableDictType_
/* Command table. sds string -> command struct pointer. */

File: src/server.c [hunk 3]
```diff
@@ -635,6 +644,15 @@ dictType hashDictType = {
     dictSdsDestructorM          /* val destructor */
 };
 
+dictType hashDictTypeZ = {
```
Comment (michalbiesek) [inline src/server.c]: add description of this structure, like for example it is done for  _commandTableDictType_
/* Command table. sds string -> command struct pointer. *

File: src/server.c [hunk 4]
```diff
@@ -635,6 +644,15 @@ dictType hashDictType = {
...
+    dictSdsHash,                /* hash function */
+    NULL,                       /* key dup */
+    NULL,                       /* val dup */
+    dictSdsKeyCompare,          /* key compare */
+    dictSdsDestructor,         /* key destructor */
```
Comment (michalbiesek) [inline src/server.c]: fix align here and line below

File: src/t_set.c [hunk 5]
```diff
@@ -239,7 +240,7 @@ void setTypeConvertA(robj *setobj, int enc, alloc a) {
 
     if (enc == OBJ_ENCODING_HT) {
         int64_t intele;
-        dict *d = dictCreate(&setDictType,NULL);
+        dict *d = dictCreate((a->alloc == z_alloc->alloc ? &setDictTypeZ : &setDictTypeM),NULL);
```
Comment (michalbiesek) [inline src/t_set.c:243]: I think better solution for this is to add comparator for whole **alloc** structure - You could do this in separate PR - intention of this comment is to mark the place of compare operation occurrence, same for https://github.com/KFilipek/redis/blob/e149bb6a23d0ca112ea0108345eeb9bb4385271e/src/object.c#L1093

Comment (KFilipek) [reply src/t_set.c:243]: Okey, I will do this change in separate PR.

**Comment by michalbiesek** [comment]
LGTM

**Review by michalbiesek — APPROVED** [review]
